packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -811,13 +811,24 @@ const NAME_TAG: string = 'Name';
  */
 export interface VpcProps {
 
+  /**
+   * The Ipam Provider to use to allocate IP Space to your VPC.
+   *
+   * Options include static allocation or from a pool.
+   *
+   * @default ipam.StaticIpam
+   */
+  readonly ipamProvider?: IIpamProvider;
```

Comment:
After discussion with a few people.

- preference for vpc prop and factory class to be the same to help find what your looking for.
- name should include some concept that it deals with ip space and the allocation management.
- avoid using the ipam for vpc prop as its loaded with assumption and is already used in awsIpam.
- no need to have Provider added to the name if the name is better.

change to: 
```ts
ipAddressManager: IpAddressManager.cidr('1.2.3.4/16'),
```
```ts
ipAddressManager: IpAddressManager.awsIpam(myIpam),
```

Reply:
@mrpackethead @juhala-aws @connelldave
Having had a few bits of further feedback any further thoughts:

Some of the leading running names from discussion for the prop and the factory are:

```ts
ipAddressManager: IpAddressManager.cidr('1.2.3.4/16')

ipAddressManager: IpAddressManager.awsIpam({..})
```

```ts
IpAddresses: IpAddressManager.cidr('1.2.3.4/16')

IpAddresses: IpAddressManager.awsIpam({..})
```

```ts
IpAddresses: IpAddresses.cidr('1.2.3.4/16')

IpAddresses: IpAddresses.awsIpam({..})
```

Reply:
> ```ts
> IpAddresses: IpAddresses.cidr('1.2.3.4/16')
> 
> IpAddresses: IpAddresses.awsIpam({..})
> ```

I think this one is my preference - encapsulating and exposing NetworkBuilder is going to be a bit of a confusion point for people used to previous behaviour, but this option smooths that bump out a little, rather than wondering what new behaviour an IpAddressManager brings by default (ie none).

Reply:
I'm largely with @connelldave here.. Except, that that I'm not sure that calling it 
```IpAddresses.awsIpam```   is helpful.

What this is, is an allocation from an Ipam Pool..      there is already a EC2.CfnIPAM, which creates an instance of IPAM.  I'd suggest looking at the L1 Cfn Types and probably creating some L2 types to mirror them.      In this case, think it should probably be called 'awsIpamAllocation'.